app/models/mixins/retirement_mixin.rb

```diff
@@ -39,6 +39,10 @@ def set_retirement_requester(obj_ids, requester)
     end
   end
 
+  def make_retire_request(requester_id)
+    self.class.make_retire_request(id, User.find(requester_id))
```

Comment:
@d-m-u do you need to handle `initiated_by: 'user'` vs `'system'`?

Reply:
I wasn't of the opinion that it made any sense to have the system context functionality exposed here since it tends to only for UI-initiated future VM retirement.